pkg/controllers/sync/resource.go

```diff
@@ -197,21 +197,6 @@ func (r *federatedResource) ObjectForCluster(clusterName string) (*unstructured.
 		obj.SetFinalizers(nil)
 	}
 
-	// Record the annotation/label keys in the template so we can diff it against the cluster object during retention
-	// to determine whether an annotation/label has been deleted from the template.
-	if _, err := annotationutil.AddAnnotation(
-		obj, common.TemplateAnnotationKeys,
```

Comment:
Let's rename `TemplateAnnotationKeys` to `PropagatedAnnotationKeys` since it's not just the template we're recording anymore.